pkg/codegen/pcl/binder_component.go

```diff
@@ -159,6 +159,7 @@ func ComponentProgramBinderFromFileSystem() ComponentProgramBinder {
 					diagnostics = diagnostics.Append(errorf(nodeRange, err.Error()))
 					return nil, diagnostics, err
 				}
+				defer file.Close()
```

Comment:
It's very annoying Go doesn't have a way to do scoped defers because these files can actually be closed as soon as ParseFile returns. `defer` will keep them open till the end of the whole function.

Wonder if it would be better to just do a `contract.IgnoreError(file.Close())` after the ParseFile call?